File: frontend/src/types/chunk.ts
export type ChunkStrategy =
  | 'fixed'
  | 'sliding_window'
  | 'recursive'
  | 'paragraph'
  | 'sentence'
  | 'markdown'
  | 'html'
  | 'semantic'
  | 'parent_child'
  | 'hierarchical'
  | 'adaptive';

export const CHUNK_STRATEGIES: { value: ChunkStrategy; label: string }[] = [
  { value: 'recursive', label: 'Recursive (default)' },
  { value: 'fixed', label: 'Fixed size' },
  { value: 'sliding_window', label: 'Sliding window' },
  { value: 'paragraph', label: 'Paragraph' },
  { value: 'sentence', label: 'Sentence' },
  { value: 'markdown', label: 'Markdown (structural)' },
  { value: 'html', label: 'HTML (structural)' },
  { value: 'semantic', label: 'Semantic' },
  { value: 'parent_child', label: 'Parent/child' },
  { value: 'hierarchical', label: 'Hierarchical' },
  { value: 'adaptive', label: 'Adaptive' },
];

export type ChunkSetStatus = 'pending' | 'ready' | 'failed';
export type ChunkStatus = 'ready' | 'failed' | 'skipped';

export interface ChunkSet {
  id: string;
  document_id: string;
  version: number;
  strategy: string;
  config: Record<string, unknown>;
  status: ChunkSetStatus;
  status_message: string | null;
  chunk_count: number;
  created_at: string;
  updated_at: string;
}

export interface Chunk {
  id: string;
  chunk_set_id: string;
  parent_chunk_id: string | null;
  chunk_index: number;
  text: string;
  char_start: number;
  char_end: number;
  token_count: number;
  page: number | null;
  heading: string | null;
  language: string | null;
  status: ChunkStatus;
  status_message: string | null;
  embedding_model: string | null;
}

export interface ChunkSetComparison {
  strategy_a: ChunkSet;
  chunks_a: Chunk[];
  strategy_b: ChunkSet;
  chunks_b: Chunk[];
}

File: backend/app/schemas/chunk.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.chunk import ChunkSetStatus, ChunkStatus


class ChunkSetRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    document_id: uuid.UUID
    version: int
    strategy: str
    config: dict
    status: ChunkSetStatus
    status_message: str | None
    chunk_count: int
    created_at: datetime
    updated_at: datetime


class ChunkRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    chunk_set_id: uuid.UUID
    parent_chunk_id: uuid.UUID | None
    chunk_index: int
    text: str
    char_start: int
    char_end: int
    token_count: int
    page: int | None
    heading: str | None
    language: str | None
    status: ChunkStatus
    status_message: str | None
    embedding_model: str | None


class GenerateChunksRequest(BaseModel):
    strategy: str = Field(..., min_length=1, max_length=20)
    config: dict | None = None


class ChunkSetComparison(BaseModel):
    strategy_a: ChunkSetRead
    chunks_a: list[ChunkRead]
    strategy_b: ChunkSetRead
    chunks_b: list[ChunkRead]
